cfgov/mega_menu/jinja2tags.py

```diff
@@ -16,11 +16,14 @@ def select_menu_for_context(context):
         except Menu.DoesNotExist:
             pass
 
-    # Next try to find a menu for the default Django language.
-    try:
-        return Menu.objects.get(language=settings.LANGUAGE_CODE[:2])
-    except Menu.DoesNotExist:
-        pass
+    # Next try to find a menu for the default Django language, if it differs
+    # from the language in the context.
+    default_language = settings.LANGUAGE_CODE[:2]
+    if default_language != language:
+        try:
+            return Menu.objects.get(language=default_language)
+        except Menu.DoesNotExist:
+            pass
```

Comment:
The changes you have here are fine, and no action is necessary on this comment, but I love nothing about this function. I wonder if there's a way to avoid repeating the try/except.

Reply:
There is, thanks to the code snippet you shared with me and I've pushed as a commit to this PR!